src/exporters/file/task.rs

```diff
@@ -0,0 +1,458 @@
+        SpanRow::from_resource_spans(resource_spans)
+    }
+
+    fn convert_metrics(&self, resource_metrics: &ResourceMetrics) -> Result<Vec<Self::MetricData>> {
+        MetricRow::from_resource_metrics(resource_metrics)
+    }
+
+    fn convert_logs(&self, resource_logs: &ResourceLogs) -> Result<Vec<Self::LogData>> {
+        LogRecordRow::from_resource_logs(resource_logs)
+    }
+
+    fn export_spans(&self, data: &[Self::SpanData], path: &Path) -> Result<()> {
+        self.export_span_rows(data, path)
+    }
+
+    fn export_metrics(&self, data: &[Self::MetricData], path: &Path) -> Result<()> {
+        self.export_metric_rows(data, path)
+    }
+
+    fn export_logs(&self, data: &[Self::LogData], path: &Path) -> Result<()> {
+        self.export_log_record_rows(data, path)
+    }
+
+    fn file_extension(&self) -> &'static str {
+        ".parquet"
+    }
+}
+
+/// Implementation of TypedFileExporter for JsonExporter
+impl TypedFileExporter for JsonExporter {
+    type SpanData = ResourceSpans;
+    type MetricData = ResourceMetrics;
+    type LogData = ResourceLogs;
+
+    fn convert_spans(&self, resource_spans: &ResourceSpans) -> Result<Vec<Self::SpanData>> {
+        Ok(vec![resource_spans.clone()])
+    }
+
+    fn convert_metrics(&self, resource_metrics: &ResourceMetrics) -> Result<Vec<Self::MetricData>> {
+        Ok(vec![resource_metrics.clone()])
+    }
+
+    fn convert_logs(&self, resource_logs: &ResourceLogs) -> Result<Vec<Self::LogData>> {
+        Ok(vec![resource_logs.clone()])
+    }
+
+    fn export_spans(&self, data: &[Self::SpanData], path: &Path) -> Result<()> {
+        self.export_traces(data, path)
+    }
+
+    fn export_metrics(&self, data: &[Self::MetricData], path: &Path) -> Result<()> {
+        self.export_metrics(data, path)
+    }
+
+    fn export_logs(&self, data: &[Self::LogData], path: &Path) -> Result<()> {
+        self.export_logs(data, path)
+    }
+
+    fn file_extension(&self) -> &'static str {
+        ".json"
+    }
+}
+
+pub async fn run_file_exporter(
+    config: FileExporterConfig,
+    traces_rx: BoundedReceiver<Vec<ResourceSpans>>,
+    metrics_rx: BoundedReceiver<Vec<ResourceMetrics>>,
+    logs_rx: BoundedReceiver<Vec<ResourceLogs>>,
+    token: CancellationToken,
+) -> Result<()> {
+    let format = config.format;
+    let path = config.path;
+    let flush_interval = config.flush_interval;
+
+    // Create output directories if they don't exist
+    let traces_dir = path.join("spans");
+    let metrics_dir = path.join("metrics");
+    let logs_dir = path.join("logs");
+
+    std::fs::create_dir_all(&traces_dir).map_err(FileExporterError::Io)?;
+    std::fs::create_dir_all(&metrics_dir).map_err(FileExporterError::Io)?;
+    std::fs::create_dir_all(&logs_dir).map_err(FileExporterError::Io)?;
+
+    info!(
+        "File exporter started, writing {} data to {}",
+        format,
+        path.display()
+    );
+
+    match format {
+        FileExporterFormat::Parquet => {
```

Comment:
The Parquet branch here and the JSON one below look identical aside from creating the exporter. Could you do something like:
```rust
let exporter = match format {
  FileExporterFormat::Parquet => std::sync::Arc::new(ParquetExporter::new()),
  FileExporterFormat::Json => std::sync::Arc::new(JsonExporter::new())
}
```
Then the remaining code would be a single implementation?